src/batch/batch.ts

```diff
@@ -1,25 +1,85 @@
 import { MessageType } from "../types/DSNP";
 import { NotImplementedError } from "../utilities";
+import request from "request";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const parquet = require("@unfinishedlabs/parquetjs");
+const { ParquetReader } = parquet;
+
+type ReadRowFunction = {
+  (row: MessageType): void;
+};
+
+interface SplitBlockBloomFilter {
+  check(value: number | string): boolean;
+}
+
+interface BloomFilterData {
+  sbbf: SplitBlockBloomFilter;
+  columnName: string;
+  RowGroupIndex: number;
+}
 
 export type BatchFileObject = string;
 
 /**
- * createFile() takes a series of activity pub objects and returns a Batch file
+ * createFile() takes a series of DSNP messages and returns a Batch file
  * object for publishing. This method is not yet implemented.
  *
- * @params  events  An array of DSNPEvents to include in the batch file
+ * @param events  An array of DSNPEvents to include in the batch file
  * @returns         A batch file object
  */
 export const createFile = (_events: MessageType[]): BatchFileObject => {
   throw NotImplementedError;
 };
 
 /**
- * openFile() takes a Batch file object . This method is not yet implemented.
+ * openURL() allows users to open a parquet file with a URL.
+ *
+ * @param url a URL to fetch parquet file from.
+ * @returns a ParquetReader object.
+ **/
+export const openURL = async (url: URL): Promise<typeof ParquetReader> => ParquetReader.openUrl(request, url);
+
+/**
+ * openFile() allows users to open a parquet file with a path.
+ *
+ * @param path path to parquet file.
+ * @returns a ParquetReader object.
+ **/
+export const openFile = async (path: string): Promise<typeof ParquetReader> => ParquetReader.openFile(path);
+
+/**
+ * readFile() reads a Parquet file by row.
  *
- * @params  file  A batch file object to open
- * @returns       An array of DSNPEvents from the batch file
+ * @param reader a ParquetReader object.
+ * @returns void.
  */
-export const openFile = (_file: BatchFileObject): MessageType[] => {
-  throw NotImplementedError;
+export const readFile = async (reader: typeof ParquetReader, callback: ReadRowFunction): Promise<void> => {
```

Comment:
It is more complex than a callback function, but do we want to have this be an Iterable?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

Reply:
the reader is already iterable.